### ISSUE TITLE
Improvements to ConfidentialLedger samples

### ISSUE THREAD

**Issue by annelo-msft** [Azure/azure-sdk-for-net#34085]
There is interest in updating the ConfidentialLedger Hello World sample in some of the following ways:
- Sleep in loops: https://github.com/Azure/azure-sdk-for-net/pull/33931#discussion_r1101769029
- Remove unused variables: https://github.com/Azure/azure-sdk-for-net/pull/33931#discussion_r1101665297

There may also be opportunities to rename variables to better reflect their usage, for example there is a getByCollectionResponse where the ledger entry is [not gotten by collection](https://github.com/Azure/azure-sdk-for-net/pull/33931/files#diff-a8c599ce75532d7d88954f0a024bf3778e3553c304c1f691638acd87cea7fb9dR115), and a few instances where snippet/non-snippet code may not agree on usage.

This issue tracks the work to improve this sample.

**Comment by pallavit** [comment]
Is this something we want to do after we have JsonData in Core? I am assuming we do not enhance samples with experimental types?

**Comment by annelo-msft** [comment]
> Is this something we want to do after we have JsonData in Core? I am assuming we do not enhance samples with experimental types?

Yes, this PR is currently only a proof of concept and to drive requirements and conversation.  That being said, if we wanted to make samples for DynamicJson public, we could feasibly do so for a library with the same GA timeline as DynamicJson.
